applications/CoSimulationApplication/tests/test_coupling_operations.py

```diff
@@ -141,6 +141,71 @@ def __CompareValues(self, old_data, new_data, factor):
         for old_val, new_val in zip(old_data, new_data):
             self.assertAlmostEqual(old_val*factor, new_val)
 
+class TestConversionOperation(KratosUnittest.TestCase):
+
+    def test_elemental_to_nodal_conversion(self):
+        self.model = KM.Model()
+        self.model_part = self.model.CreateModelPart("default")
+        self.model_part.AddNodalSolutionStepVariable(KM.FORCE)
+        self.model_part.ProcessInfo[KM.DOMAIN_SIZE] = 3
+        props = self.model_part.CreateNewProperties(1)
+
+        self.model_part.CreateNewNode(1, 0.0, 0.0, 0.0)
+        self.model_part.CreateNewNode(2, 1.0, 0.0, 0.0)
+        self.model_part.CreateNewNode(3, 1.0, 1.0, 0.0)
+        self.model_part.CreateNewNode(4, 0.0, 1.0, 0.0)
+
+        new_element = self.model_part.CreateNewElement("Element2D4N", 1, [1,2,3,4], props)
+        new_element.SetValue(KM.FORCE, [12.0, 8.0, 0.0])
+
+        self.model_part.CreateNewNode(5, 2.0, 0.0, 0.0)
+        self.model_part.CreateNewNode(6, 2.0, 1.0, 0.0)
+        new_element = self.model_part.CreateNewElement("Element2D4N", 2, [2,5,6,3], props)
+        new_element.SetValue(KM.FORCE, [16.0, 4.0, 0.0])
+
+        elemental_data = KM.Parameters("""{
+            "model_part_name" : "default",
+            "location"        : "element",
+            "variable_name"   : "FORCE",
+            "dimension"       : 3
+        }""")
+
+        self.interface_data = CouplingInterfaceData(elemental_data, self.model)
+
+        self.solver_wrappers = {"dummy_solver" : DummySolverWrapper({"elemental_data" : self.interface_data})}
+
+        self.solver_process_info = KM.ProcessInfo()
+
+        conversion_op_settings = KM.Parameters("""{
+            "type"           : "elemental_data_to_nodal_data",
+            "solver"         : "dummy_solver",
+            "data_name"      : "elemental_data",
+            "echo_level"     : 0
+        }""")
+
+        conversion_operation = coupling_operation_factory.CreateCouplingOperation(conversion_op_settings, self.solver_wrappers, self.solver_process_info)
+
+        conversion_operation.Check()
+
+        conversion_operation.Execute()
+
+        nodal_data_output_setting = KM.Parameters("""{
+            "model_part_name"       : "default",
+            "variable_name"         : "FORCE",
+            "location"              : "node_historical",
+            "dimension"             : 3
+        }""")
+
+        nodal_data_output = CouplingInterfaceData(nodal_data_output_setting,  self.model)
+
+        expected_nodal_values = [3, 2 ,0, 7, 3, 0, 7, 3 ,0, 3, 2 ,0, 4, 1 ,0, 4, 1 ,0 ]
+
+        self.__CheckData(expected_nodal_values, nodal_data_output.GetData())
```

Comment:
```suggestion
        self.assertVectorEqual(expected_nodal_values, nodal_data_output.GetData())
```
this should also work

Reply:
Yes, assertVectorAlmostEqual() worked.